scene/gui/tree.cpp

```diff
@@ -6320,7 +6324,15 @@ bool Tree::can_drop_data(const Point2 &p_point, const Variant &p_data) const {
 		return false;
 	}
 
-	return Control::can_drop_data(p_point, p_data);
+	bool is_drop_possible = Control::can_drop_data(p_point, p_data);
+	Tree *tree = const_cast<Tree *>(this);
+	if (is_drop_possible && theme_cache.scroll_speed > 0) {
+		tree->scrolling = true;
+		tree->set_process_internal(true);
+	} else if (!scrolling) {
+		tree->set_process_internal(false);
+	}
+	return is_drop_possible;
```

Comment:
I'm not too familiar with the Tree implementation but it seems somewhat hacky to have to `const_cast` and perform logic on Tree in `can_drop_data`. I would expect that function to just return yes or no, and not modify the behavior of a Tree instance.